### ISSUE TITLE
How to support CosFace loss function?

### ISSUE THREAD

**Issue by richardhahahaha** [Deci-AI/super-gradients#1647]
### 💡 Your Question

Some loss function such as CosFace has parameters that need to be trained and saved to checkpoint file, but the Trainer and checkpoint seem not to support it.

What is the solution?

**Comment by BloodAxe** [comment]
The solution is to check whether loss has trainable parameters and include them to optimizer if they are present and also store them in checkpoint. If you're in hurry the best win-win solution would be to make a PR with this fix and we will be happy to review it. 
Otherwise it may be done eventually, but I can't give an estimate when we can get to work on this feature.
